packages/ng-query/src/lib/tests/query.spec.ts

```diff
@@ -0,0 +1,126 @@
+import { QueryProvider, UseQuery } from '../query';
+import { TestBed } from '@angular/core/testing';
+import { fetcher, rejectFetcher, simpleFetcher } from './test-utils';
+import { QueryClient } from '../query-client';
+import { QueryClient as QueryCore } from '@tanstack/query-core';
+import { QueryObserver } from '@tanstack/query-core';
+import { baseQuery } from '../base-query';
+import { skip, firstValueFrom } from 'rxjs';
+jest.mock('../base-query');
+
+describe('useQuery', () => {
+  let useQuery: UseQuery;
+  let client: QueryCore;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    useQuery = TestBed.inject(QueryProvider);
+    client = TestBed.inject(QueryClient);
+  });
+
+  it('should properly execute query', () => {
+    useQuery(['key0'], simpleFetcher, { staleTime: 1000 });
+    expect(baseQuery).toBeCalledWith(client, QueryObserver, {
+      staleTime: 1000,
+      queryKey: ['key0'],
+      queryFn: expect.any(Function), // TODO: make it work with simpleFetcher function
+    });
+  });
+
+  it('should return loading status initially', () => {
+    const query = useQuery(['key1'], simpleFetcher);
+
+    expect(query.getCurrentResult()).toMatchObject({
+      status: 'loading',
+      isLoading: true,
+      isFetching: true,
+      isStale: true,
+    });
+  });
+
+  it('should resolve to success: useQuery(key, dataFn)', async () => {
+    const query = useQuery(['key2'], fetcher('result2'));
+
+    const result = await firstValueFrom(query.result$.pipe(skip(1)));
+    expect(result).toMatchObject({
+      status: 'success',
+      data: 'result2',
+      isLoading: false,
+      isFetching: false,
+      isFetched: true,
+      isSuccess: true,
+    });
+
+    /* expect(query.result$).toBeObservable( */
```

Comment:
@NetanelBasal  I wanted to have marble tests in here using `jest-marbles`, but `query.result$` emits only 1 frame. Could you have a look please ?

Reply:
I'd use https://github.com/hirezio/observer-spy

Reply:
let me try it.

Reply:
oh this works like a charm. 🔥 marble tests are hard.